src/ng_package/text_replace/index.ts

```diff
@@ -0,0 +1,64 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import {globSync} from 'tinyglobby';
+
+/**
+ * Interface describing a file captured in the Bazel action.
+ * https://docs.bazel.build/versions/main/skylark/lib/File.html.
+ */
+export interface BazelFileInfo {
+  /** Execroot-relative path pointing to the file. */
+  path: string;
+  /** The path of this file relative to its root. e.g. omitting `bazel-out/<..>/bin`. */
+  shortPath: string;
+}
+
+/** Apply the provided substitutions to the content of the file at the origin, writing it to the provided destination. */
+async function applySubstitutions(origin: string, destination: string, substitutions: [RegExp, string][]) {
+  // Ensure that a directory exists for the destination file to be written into. We don't guarantee order of substitutions,
+  // so we need to make sure manually for the provided file paths.
+  await fs.mkdir(path.dirname(destination), {recursive: true});
+
+  /** The content of the file to the run the substitutions against. */
+  let content = await fs.readFile(origin, {encoding: 'utf-8'});  
+
+  substitutions.forEach(([regexp, replacement]) => {
+    content = content.replace(regexp, replacement);
+  });
+
+  return fs.writeFile(destination, content);
+}
+
+async function main(args: string[]) {
+  const [substitutionsArg, originsRaw, destinationRaw] = (await fs.readFile(args[0], {encoding: 'utf-8'})).split('\n').map(line => line.replace(/^'(.*)'$/, '$1'));
```

Comment:
Nit: format a bit (I know we need prettier :D)

Also what is the replace doing?

Reply:
The replace removes the additional single quotes that are placed around the inputs from the `Args.add` call in the starlark code: https://bazel.build/rules/lib/builtins/Args

Reply:
interesting.